### ISSUE TITLE
Add option to ignore errors in 3rd party libraries (e.g. PyTorch)

### ISSUE THREAD

**Issue by bfelbo** [microsoft/pyright#312]
**Is your feature request related to a problem? Please describe.**

Arguably, the most popular machine learning library right now is [PyTorch](https://pytorch.org/). When installing and importing it, pyright detects ~80 errors as shown below.

```
import torch
import torch.nn as nn
import torch.nn.functional as F
```
![image](https://user-images.githubusercontent.com/23178145/67643482-458b7280-f8d5-11e9-9743-6148b2bf28e6.png)

It's unclear to me how I'll make pyright ignore these errors. Adding the path to `pyrightconfig.json` is ignored, i.e.
```
    "ignore": [
        "/usr/local/anaconda3/envs/spirit/lib/python3.6/site-packages/torch/*.py"
    ]
``` 
results in `Ignoring path "/usr/local/anaconda3/envs/spirit/lib/python3.6/site-packages/torch/*.py" in "ignore" array because it is not relative.` in the pyright output.

**Describe the solution you'd like**
It would be great with an option to ignore errors in 3rd party libraries. I'd be surprised if this isn't also relevant for many other libraries besides PyTorch.

Ideally, it would be as simple as setting `reportExternalLibraryErrors: false`in `pyrightconfig.json`.

**Comment by erictraut** [comment]
Hmm, something is definitely going wrong here. Pyright shouldn't be attempting to analyze any third-party libraries (much less report any errors in them) unless you've installed them within one of your "include" directories. What paths do you have in the "include" array in the pyrightconfig.json file?

**Comment by bfelbo** [comment]
Yeah, I was surprised too. This is the entire pyrightconfig.json:
```
{
    "reportMissingImports": false,
    "include": ["./src/*.py"],
    "exclude": [
        "./src/representation_method.py",
        "./src/rule_based.py",
        "./src/spirit.py"
    ],
    "ignore": [
        "/usr/local/anaconda3/envs/spirit/lib/python3.6/site-packages/torch/*.py"
    ]
}
```

**Comment by bfelbo** [comment]
I guess this might have been triggered by doing “Go to definition” on something imported from PyTorch. I often have to look up the source. Hopefully there’s a way to not have that cause an analysis of the entire library codebase?

**Comment by christopherkuemmel** [comment]
I have the same issue that pyright displays errors from the PyTorch lib. This issue is may related to #273 . 

Pyright version 1.0.71 originally fixed my problem. However in the latest version, the errors occur again. Despite those are somewhat different.

**Comment by erictraut** [comment]
The reason this affects pytorch is that it ships with type stub files that import non-type stub files (implementation files). Pyright tries to avoid analyzing third-party implementations, but it will analyze third-party type stubs. In doing so, it continued to analyze the imported implementation files.

The fix for this problem will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is fixed in version 1.0.83, which I just published.

**Comment by bfelbo** [comment]
Great, thank you!
